sfs/util.py

```diff
@@ -491,3 +496,31 @@ def _count_walls_1d(a):
         wall_count = wall_count[N_mask, :]
 
     return xs, wall_count
+
+
+def spherical_hn2(n, z):
+    r"""Spherical Hankel function of 2nd kind.
+
+    Defined as http://dlmf.nist.gov/10.47.E6,
+
+    .. math::
+
+        \hankel{2}{n}{z} = \sqrt{\frac{\pi}{2z}}
+        \Hankel{2}{n + \frac{1}{2}}{z},
+
+    where :math:`\Hankel{2}{n}{\cdot}` is the Hankel function of the
+    second kind.
+
+    Parameters
+    ----------
+    n : array_like
+        Order of the spherical Hankel function (n >= 0).
+    z : complex or float, array_like
+        Argument of the spherical Hankel function.
+
+    Returns
+    -------
+    hn2 : array_like
```

Comment:
I wouldn't invent a variable name for a single return value.
I think this only helps if there are multiple return values.

I normally don't use "array_like" for return values, because those are typically more concrete types, like "numpy.ndarray".

However, in this case there are multiple possibilities ... to make this clear, we could write something like

```rst
    Returns
    -------
    complex scalar or numpy.ndarray
        The function value(s) for the given order(s) and argument(s).
```

But this seems unnecessarily verbose, doesn't it?

I think in this case we could also completely drop the whole "Returns" section, because neither the return type nor the description is particularly interesting or surprising.
The only real information in there is that the function values are complex. But if we need to say that, we can also just mention it in the main description.

Reply:
I found it also verbose, that was the reason why I skipped the description of the return value.
I removed it now completely.